src/Relations/BelongsToThrough.php

```diff
@@ -350,7 +367,11 @@ public function getThroughParents()
      */
     public function getFirstForeignKeyName()
     {
-        return $this->prefix . $this->getForeignKeyName(end($this->throughParents));
+        // TODO: the method docblock doesn't match with the usage of `end($this->throughParents)`
```

Comment:
@staudenmeir can you take a look at this?

Reply:
TLDR: There are historical reasons for this contradiction and the code is correct.

This package has been created by another developer and I took over the maintenance when he stopped developing with PHP. Due to its similarity with my `eloquent-has-many-deep` package, I try to align the code with the other package when fixing bugs or adding features. A key difference between the packages is the order of "through" models provided by the user: This package requires _related to parent model_, while it's _parent to related model_ for the other one.

I adopted the `getFirstForeignKeyName()` method from the other package and so "first" is referring to the "through" model that's closest to the parent model. In this package, however, this model is the last one in the array.

Reply:
Thanks for elaborating, then I think there aren't too many surprises in this PR anymore, although it was a bit difficult to make sure I replaced the Model types with the correct option out of `TRelatedModel`, `TIntermediateModel`, and `TDeclaringModel, so it's possible that I made a mistake somewhere there